src/es/components/controllers/autoCompleteLocation/AutoCompleteLocation.js

```diff
@@ -0,0 +1,212 @@
+
+/** @typedef {{
+  description?: string,
+  selected: string,
+}} LocationSelectionItem */
+
+/** @typedef {{
+  lat: number,
+  lng: number,
+}} LocationCoordinates */
+
+/** @typedef {{
+  total: number,
+  success: boolean,
+  searchText: string,
+  items: google.maps.places.AutocompletePrediction[] || LocationItem[],
+  cms: []
+}} fetchAutoCompleteEventDetail */
+
+/* global fetch */
+/* global AbortController */
+/* global CustomEvent */
+
+import { Shadow } from '../../web-components-toolbox/src/es/components/prototypes/Shadow.js'
+
+/**
+ * AutoComplete are retrieved via the corresponding endpoint as set as an attribute
+ * As a controller, this component communicates exclusively through events
+ * Example: web-components-toolbox-klubschule
+ *
+ * @export
+ * @class AutoCompletePredictions
+ * @type {CustomElementConstructor}
+ */
+export default class AutoCompleteLocation extends Shadow() {
+  constructor(options = {}, ...args) {
+    super({
+      importMetaUrl: import.meta.url,
+      mode: 'false',
+      ...options
+    }, ...args)
+    this.abortController = null
+    this.RESOLVE_MSG = 'LOADED'
+  }
+
+  connectedCallback() {
+    Promise
+      .resolve(this.loadDependency())
+      .then(() => {
+        this.addEventListener(this.getAttribute('request-auto-complete') || 'request-auto-complete-location', this.requestAutoCompleteListener)
```

Comment:
add event listeners immediately without waiting for the loadDependency... this could have the sideeffect, that the event is added twice. Imagine connect (not added event listener yet, waiting for Promise), disconnect (not removing the event listener since they are not connected yet, waiting for Promise), connect adding the event listener a second time